test/EventStore.Client.Streams.Tests/Append/append_to_stream.cs

```diff
@@ -423,6 +424,47 @@ await Fixture.Streams
 		ex.StatusCode.ShouldBe(StatusCode.DeadlineExceeded);
 	}
 
+	[Fact]
+	public async Task should_not_append_to_stream_when_error_thrown_midway() {
+		var       streamName            = Fixture.GetStreamName();
+		const int initialNumberOfEvents = 5;
+
+		// Append some events before
+		await Fixture.Streams.AppendToStreamAsync(
+			streamName,
+			StreamState.Any,
+			Fixture.CreateTestEvents(initialNumberOfEvents)
+		);
+
+		// Force regular append by passing credentials
+		await Assert.ThrowsAsync<EnumerationFailedException>(
+			async () =>{
+				await Fixture.Streams.AppendToStreamAsync(
+					streamName,
+					StreamState.StreamExists,
+					GetEvents(),
+					userCredentials: new UserCredentials("admin", "changeit")
+				);
+			}
+		);
+
+		// No more events should be appended to the stream
+		var result = Fixture.Streams.ReadStreamAsync(Direction.Forwards, streamName, StreamPosition.Start);
+		var eventsCount = await result.CountAsync();
```

Comment:
can we do single call here?
`var eventsCount = Fixture.Streams.ReadStreamAsync(Direction.Forwards, streamName, StreamPosition.Start).CountAsync()`